app/models/course.rb

```diff
@@ -226,4 +226,20 @@ def publish_enrichment(current_user)
       enrichment.publish(current_user)
     end
   end
+
+  def toggle_site(site_to_toggle)
+    is_course_new = new? # persist this before we change anything
+
+    site_status = site_statuses.find_or_initialize_by(site: site_to_toggle)
+    if !site_status.new_record?
+      site_status.toggle_status!
+    else
+      site_status.save!
+      site_status.start! unless is_course_new
+    end
+  end
```

Comment:
This isn't very clear to me wrt what it's actually doing. Discussed IRL so I'll leave it with you. :)

Reply:
I've replaced `toggle` with `add_site!` and `remove_site!`